client/src/pages/TeacherView.jsx

```diff
@@ -0,0 +1,127 @@
+import React, {useEffect} from "react";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import {useLocation} from "react-router-dom";
+
+import "./teacher.css"
+function TeacherView() {
+    /*For the rendering on the same page */
+    const [selectedCourse, setSelectedCourse] = useState(null);
+    const [courses, setCourses] = useState([]);
+    const location = useLocation();
+    /*Defining the classes- for now it's hardcoded, will have to connect to the db (will have to fetch courses by the API and use json)*/ 
+    
+    /*const classes=[
+        {name:"SOEN 341"},
+        {name:"ELEC 275"},
+        {name:"COMP 123"},
+        {name: "ENGR 232"}
+    ];*/
+
+    //Fetch the courses
+    useEffect(() => {
+        const fetchCourses = async() =>{
+            try{
+                const profId = "PROF ID HERE";
+                const response = await fetch(`/teacher/courses?prof_id=${profId}`);
+                const data = await response.json();
+                setCourses(data);
+            }
+            catch(error){
+                console.error("Error");
+            }
+        };
+        fetchCourses();
+    }, []);
+
+    const handleMouseMove = (e) => {
+        const box = e.currentTarget;
+        const rect = box.getBoundingClientRect();
+        const x = e.clientX - rect.left;
+        const y = e.clientY - rect.top;
+        box.style.setProperty('--x', `${x}px`);
+        box.style.setProperty('--y', `${y}px`);
+      };
+    
+      const handleMouseLeave = (e) => {
+        const box = e.currentTarget;
+        box.style.setProperty('--x', `-9999px`);
+        box.style.setProperty('--y', `-9999px`);
+      };
```

Comment:
I believe that here you are trying to conditionally render something when something is hovered on. Please do not use `getBoundingClientRect` or `setProperty` as the latter could lead to security issues.

What i suggest is use the `onHover` option of the component and that calls a function called `handleHover` that sets a boolean variable using useState and then that gets conditionally rendered.

Reply:
Got it! Thank you for the feedback :)). I have fixed it and will commit my changes.